lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/LeastSquaresExpansion.cxx

```diff
@@ -0,0 +1,195 @@
+
+CLASSNAMEINIT(LeastSquaresExpansion)
+
+static const Factory<LeastSquaresExpansion> Factory_LeastSquaresExpansion;
+
+
+/* Default constructor */
+LeastSquaresExpansion::LeastSquaresExpansion()
+  : FunctionalChaosAlgorithm()
+{
+  // Nothing to do
+}
+
+
+/* Constructor */
+LeastSquaresExpansion::LeastSquaresExpansion(const Sample & inputSample,
+    const Sample & outputSample,
+    const Distribution & distribution)
+  : LeastSquaresExpansion(inputSample, Point(inputSample.getSize(), 1.0), outputSample, distribution)
+{
+  // Nothing to do  
+}
+
+/* Constructor */
+LeastSquaresExpansion::LeastSquaresExpansion(const Sample & inputSample,
+    const Point & weights,
+    const Sample & outputSample,
+    const Distribution & distribution)
+  : FunctionalChaosAlgorithm(inputSample, weights, outputSample, distribution)
+{
+  // Extract the basis from the adaptive strategy attribute of the base class
+  basis_ = getAdaptiveStrategy().getBasis();
+  // The basis size is the maximum dimension of the adaptive strategy
+  basisSize_ = getAdaptiveStrategy().getMaximumDimension();
+  // The method name is set to the default one, given by ResourceMap
+  methodName_ = ResourceMap::GetAsUnsignedInteger("FunctionalChaosAlgorithm-LeastSquaresMethod");
+}
+
+/* Constructor */
+LeastSquaresExpansion::LeastSquaresExpansion(const Sample & inputSample,
+    const Sample & outputSample,
+    const Distribution & distribution,
+    const OrthogonalBasis & basis,
+    const UnsignedInteger basisSize,
+    const String & methodName)
+  : LeastSquaresExpansion(inputSample, Point(inputSample.getSize(), 1.0), outputSample, distribution, basis, basisSize, methodName)
+{
+  // Nothing to do  
+}
+
+/* Constructor */
+LeastSquaresExpansion::LeastSquaresExpansion(const Sample & inputSample,
+    const Point & weights,
+    const Sample & outputSample,
+    const Distribution & distribution,
+    const OrthogonalBasis & basis,
+    const UnsignedInteger basisSize,
+    const String & methodName)
+  : FunctionalChaosAlgorithm(inputSample, weights, outputSample, distribution)
+  , basis_(basis)
+  , basisSize_(basisSize)
+  , methodName_(methodName)
+{
+  // The arguments are checked in the base class, excepted the basis and its size
+  if (basis.getMeasure().getDimension() != distribution.getDimension()) throw InvalidArgumentException(HERE) << "Error: the basis must have a measure with the same dimension as the input distribution, heare measure dimension=" << basis.getMeasure().getDimension() << " and distribution dimension=" << distribution.getDimension();
+  if (basisSize == 0) throw InvalidArgumentException(HERE) << "Error: cannot project on a basis of size zero";
+}
+
+
+/* Virtual constructor */
+LeastSquaresExpansion * LeastSquaresExpansion::clone() const
+{
+  return new LeastSquaresExpansion(*this);
+}
+
+
+/* String converter */
+String LeastSquaresExpansion::__repr__() const
+{
+  return OSS() << "class=" << getClassName();
```

Comment:
I suggest to `repr` the attributes of the `LeastSquaresExpansion` class, that is `basisSize_`, `basis_` and `methodName_` in that particular order.

Reply:
`__repr__` duplicated L 190

Reply:
Is that fixed?